src/tests/performance/connections.rs

```diff
@@ -170,15 +170,34 @@ async fn p002_connections_load() {
         .unwrap();
     let node_addr = node.addr();
 
-    // This is "the hack" but is needed to perform next tests if IPS table is not empty. The
-    // standard TIME_WAIT is 60s before we can use the same addr and port again.
-    // So we're taking already used IPs and each thread in each iteration will use different IP.
-    // If the table is empty or too small, the thread itself will notice it and will use the
-    // local IP.
-    // It can be removed once pea2pea will offer REUSE_ADDR options.
-    let mut ip_idx = 0;
+    let mut port_idx = 0;
 
     for synth_count in synth_counts {
+        let mut synth_sockets = Vec::with_capacity(synth_count.into());
+        for i in 0..synth_count {
+            let socket = TcpSocket::new_v4().unwrap();
+
+            // Make sure we can reuse the address and port
+            socket.set_reuseaddr(true).unwrap();
+            socket.set_reuseport(true).unwrap();
+
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            if IPS.len() > i as usize {
+                let source_addr = SocketAddr::new(
+                    IpAddr::V4(Ipv4Addr::from_str(IPS[i as usize]).unwrap()),
+                    CONNECTION_PORT + port_idx,
+                );
+                port_idx += 1;

```

Comment:
Shouldn't we do this in the outer for loop (which starts with line 175) 🤔

Reply:
Or why do we need to change port_idx at all?
Didn't we say that restarting the rippled node after every subtest helps with the reuseaddr issue?

Reply:
Changed to restart node each subset iteration. port_idx is eliminated.